src/site/content/en/blog/web-codecs/index.md

```diff
@@ -0,0 +1,361 @@
+for decoders.
+
+```js
+let responses = await downloadVideoChunksFromServer(timestamp);
+for (let i = 0; i < responses.length; i++) {
+  let chunk = new EncodedVideoChunk({
+    timestamp: responses[i].timestamp,
+    data: new Uint8Array ( responses[i].body )
+  });
+  decoder.decode(chunk);
+}
+await decoder.flush();
+```
+
+Now it's time to show how a freshly decoded frame can be shown on the page. It's
+better to make sure that the decoder output callback  (`handleFrame()`)
+quickly returns. In the example below, it only adds a frame to the queue of
+frames ready for rendering.
+Rendering happens separately, and consists of three steps:
+
+1.  Converting the `VideoFrame` into an
+    [`ImageBitmap`](https://developer.mozilla.org/en-US/docs/Web/API/ImageBitmap).
+1.  Waiting for the right time to show the frame.
+1.  Drawing the image on the canvas.
+
+Once a frame is no longer needed, call `destroy()` to release underlying memory
+before the garbage collector gets to it, this will reduce the average amount of
+memory used by the web application.
+
+```js
+let cnv = document.getElementById("canvas_to_render");
+let ctx = cnv.getContext("2d", { alpha: false });
+let ready_frames = [];
+let underflow = true;
+let time_base = 0;
+
+function handleFrame(frame) {
+  ready_frames.push(frame);
+  if (underflow)
+    setTimeout(render_frame, 0);
+}
+
+function delay(time_ms) {
+  return new Promise((resolve) => {
+    setTimeout(resolve, time_ms);
+  });
+}
+
+function calculateTimeTillNextFrame(timestamp) {
+  if (time_base == 0)
+    time_base = performance.now();
+  let media_time = performance.now() - time_base;
+  return Math.max(0, (timestamp / 1000) - media_time);
+}
+
+async function render_frame() {
+  if (ready_frames.length == 0) {
+    underflow = true;
+    return;
+  }
+  let frame = ready_frames.shift();
+  underflow = false;
+
+  let bitmap = await frame.createImageBitmap();
+  frame.destroy();
+  // Based on the frame's timestamp calculate how much of real time waiting
+  // is needed before showing the next frame.
+  let time_till_next_frame = calculateTimeTillNextFrame(frame.timestamp);
+  await delay(time_till_next_frame);
+  ctx.drawImage(bitmap, 0, 0);
+
+  // Immediately schedule rendering of the next frame
+  setTimeout(render_frame, 0);
+}
+```
+
+## Demo
+
+The demo below shows two canvases, the first one is animated at the refresh rate
+of your display, the second one shows a sequence of frames captured by
+`VideoTrackReader` at 30fps, encoded  and decoded using WebCodecs API.
+
+[https://webcodecs-blogpost-demo.glitch.me/](https://webcodecs-blogpost-demo.glitch.me/)
+
+## Feature Detection
+
+This is an easy way to check for WebCodecs support, to make  sure that the
+origin trial or the flag took effect.
+
+```js
+if ("VideoEncoder" in window) {
+  // WebCodecs API is supported.
+}
+```
+
+## Using the WebHID API {: #use }
+
+### Enabling via chrome://flags
```

Comment:
There is no way to enable WebCodecs via chrome://flags. 
Please preserve original wording:

```
To enable WebCodecs locally, start Chrome with a command line flag: --enable-blink-features=WebCodecs.
```

Reply:
Fixed.